### PR TITLE
Additionally check file mimetype after decryption

### PR DESCRIPTION
**Pull request by anoadragon453**
Before this change, we would only check the mimetype of an encrypted file based on the information provided with the file by the client.

This change now additionally checks the mimetype of encrypted files based on the file contents (after it has been decrypted) - just as we do for non-encrypted files.

### CODE REVIEW

**Review by richvdh — COMMENTED** [review]
I've not really looked at this project before, but this looks ... plausible?

Is there a particular bug we're trying to fix? What would be the user-visible effect of that bug?

**Comment by anoadragon453** [comment]
Nothing will change much from the user's perspective. This is just aligning mimetype checks across both encrypted and non-encrypted files, whereas before the two paths were a bit inconsistent with each other.

File: src/reporting.js [hunk 1]
```diff
@@ -314,23 +314,44 @@ async function generateReport(console, httpUrl, matrixFile, filePath, tempDir, s
...
+
+        // Further validate the mimetype of the file from the decrypted content
+        let mimetype = fileType(decryptedFileContents);
+        if (mimetype === null) {
+            return {clean: false, info: 'File type not supported'};
+        } else if (mimetypeArray && !mimetypeArray.includes(mimetype)) {
+            return {clean: false, info: 'File type not supported'};
+        }
+
+        // Write the decrypt file bytes to disk
```
Comment (richvdh) [inline src/reporting.js:340]:
```suggestion
        // Write the decrypted file bytes to disk
```

**Review by richvdh — APPROVED** [review]
lgtm

**Pull request merged**